server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -1684,26 +1685,40 @@ public Boolean providesDerived(Pool pool, String derivedProvidedProductId) {
             }
         }
 
-        return this.getProvidedProductIdsByPoolIds(poolIds);
+        return this.getProvidedProductByPoolIds(poolIds, true);
     }
 
     /**
-     * Fetches a mapping of pool IDs to sets of product IDs representing the provided products of
+     * Fetches a mapping of pool IDs to sets of product IDs or UUIDs representing the provided products of
      * the given pool. The returned map will only contain mappings for pools specified in the given
      * collection of pool IDs.
      *
      * @param poolIds
-     *  A collection of pool IDs for which to fetch provided product IDs
+     *  A collection of pool IDs for which to fetch provided product IDs or UUIDs
+     *
+     * @param fetchMode
```

Comment:
Nack on this.

This completely changes the typing on our output, which is *very* bad. Consistency is more important than code reuse.

Reply:
Removed this piece of code from here.